Support __doc__ argument in create_model()

chris-spann opened the pull request:
<!-- Thank you for your contribution! -->
<!-- Unless your change is trivial, please create an issue to discuss the change before creating a PR -->

## Change Summary
Fixes runtime error when using `__doc__` argument for `create_model()`
<!-- Please give a short summary of the changes. -->

## Related issue number

<!-- WARNING: please use "fix #123" style references so the issue is closed when this PR is merged. -->
fix [#7481](https://github.com/pydantic/pydantic/issues/7841)

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [x] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**


Selected Reviewer: @samuelcolvin

chris-spann commented:
please review

sydney-runkle commented:
Hi @chris-spann,

Thanks so much for your contribution! ⭐ 

I think we'll want to add support explicitly for the `__doc__` argument to the `create_model` function by modifying the signature of the `create_model` function in `main.py` and assigning the docstring of the new class within the `create_model` function as well.

Please update 👍

chris-spann commented:
Thank you @sydney-runkle for the review! Please let me know if I've adequately addressed your comment in [my latest commit](https://github.com/pydantic/pydantic/pull/7863/commits/a7ead6800edf29487789ea50d4258ca8364293e7).

sydney-runkle commented on tests/test_main.py:
```diff
@@ -59,6 +59,20 @@ class Model(BaseModel):
     m = Model(a=10.2)
     assert m.a == 10.2
     assert m.b == 10
+    assert m.__doc__ is None
+
+
+def test_success_docstring():
+    class Model(BaseModel):
+        """Another model"""
+
+        a: float
+        b: int = 10
+
+    m = Model(a=10.2)
+    assert m.a == 10.2
+    assert m.b == 10
+    assert m.__doc__ == 'Another model'
```
We can leave out this addition to the tests, as nothing here is affected by the changes to `create_model`

chris-spann replied:
Thank you! I have addressed this [here](https://github.com/pydantic/pydantic/pull/7863/commits/291dd3ad68f3fb1b6023acc6bd36d868bd717824).

sydney-runkle commented on tests/test_create_model.py:
```diff
@@ -254,15 +254,22 @@ def test_create_model_field_and_model_title():
+    assert model.__name__ == 'FooModel'
+    assert model.__doc__ == 'The Foo model'
+
+
```
Awesome, thanks for adding some tests. Could you add one more test that checks that if a model is created with a `__base__` (some other model), the new model inherits the docstring from the `__base__` model?

chris-spann replied:
No problem! I have added a test for this [here](https://github.com/pydantic/pydantic/pull/7863/commits/291dd3ad68f3fb1b6023acc6bd36d868bd717824).

sydney-runkle reviewed (changes requested):
LGTM other than a few small changes I left comments for.

Once you add those small changes, I'll approve and merge.

Thanks again for your contribution. This looks great 🌟

sydney-runkle reviewed (approved):
LGTM, thanks for the contribution! 🎈
